### ISSUE TITLE
detect-virt can't detect parallels virtual machine

### ISSUE THREAD

**Issue by catap** [systemd/systemd#23856]
### systemd version the issue has been seen with

250

### Used distribution

NixOS

### Linux kernel version used

5.15.49

### CPU architectures issue was seen on

x86_64

### Expected behaviour you didn't see

```sh
[nix-shell:~]$ systemd-detect-virt
parallels
```

### Unexpected behaviour you saw

```sh
[nix-shell:~]$ systemd-detect-virt
kvm
```

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
[nix-shell:~]$ sudo dmidecode -s system-product-name
Parallels Virtual Platform

[nix-shell:~]$ systemd-detect-virt
kvm

[nix-shell:~]$ sudo virt-what
parallels

[nix-shell:~]$
```

**Comment by yuwata** [comment]
Could you show the contents of the files below?
https://github.com/systemd/systemd/blob/d5548eb618d426a3952746eb3a0e7d95d186ea7e/src/basic/virt.c#L147-L151

**Comment by catap** [comment]
@yuwata,
```sh
[nix-shell:~]$ cat /sys/class/dmi/id/product_name
Parallels Virtual Platform

[nix-shell:~]$ cat /sys/class/dmi/id/sys_vendor
Parallels Software International Inc.

[nix-shell:~]$ cat /sys/class/dmi/id/board_vendor
Parallels Software International Inc.

[nix-shell:~]$ cat /sys/class/dmi/id/bios_vendor
Parallels Software International Inc.

[nix-shell:~]$ cat /sys/class/dmi/id/product_version
None

[nix-shell:~]$ 
```

**Comment by yuwata** [comment]
@catap Thanks. Fix is waiting in #23859. Could you test the PR?
